src/emulator/pubsubEmulator.ts

```diff
@@ -16,14 +19,19 @@ export interface PubsubEmulatorArgs {
   auto_download?: boolean;
 }
 
+interface Trigger {
+  triggerKey: string;
+  signatureType: SignatureType;
+}
+
 export class PubsubEmulator implements EmulatorInstance {
   pubsub: PubSub;
 
   // Map of topic name to a list of functions to trigger
-  triggers: Map<string, Set<string>>;
+  triggersByTopic: Map<string, { triggerKeys: Set<string>; triggers: Trigger[] }>;
 
   // Map of topic name to a PubSub subscription object
-  subscriptions: Map<string, Subscription>;
+  subscriptionsByTopic: Map<string, Subscription>;
```

Comment:
Consider renaming to `subscriptionByTopic` (drop `s` since there is only one subscription per topic)